### PR TITLE
Sorting Algorithms

### PR DESCRIPTION
**Pull request by divya1509**
Added a MicroByte to involve the learners to help them understand Sorting Algorithms.

Checklist:
- [x] Added activities for learners
- [x]  Didn't give the solutions to the added activities straight away
- [x] Followed a LearnByDoing approach
- [x]  Clean and well-documented code (if source code files are involved)

### CODE REVIEW

**Review by amoghrajesh — CHANGES_REQUESTED** [review]
Hey @divya1509!
This is wholesome content with a lot of breadth coverage.
However, I would want to request a few changes to the content:

1. The pseudo code for Bubble Sort is extremely brief, you can make it more detailed.
2. Provide links pertaining to each of the algorithms so that learners go through it and later come to see your code.
3. Try to include some images/gif's for easier understanding because this is just plain text which might get hard to read all at once.
4. You can add a section on when to use which one in form of questions for learners to make it more engaging (Probably give out a scenario like you want to sort a linked list, which algo is most suitable?)
5. If possible, include Selection Sort as it is the most basic one and should probably be the first one (Maybe provide some relatable approach like you have given for insertion sort - 'deck of cards'. For selection sort, you can probably say something like how we get ourselves arranged for school assembly in height wise order - pick the smallest one and bringing him to start right?)

Apart from all these small changes, great content!
Try to add some activities to make it more engaging too 😄 
Make these changes and we will re-review ASAP.
Also, after done drop a comment saying Done or something so we don't miss it out.

Thanks for contributing!

**Comment by divya1509** [comment]
I have done the required changes. Please check them.

**Review by amoghrajesh — CHANGES_REQUESTED** [review]
Hey @divya1509!
Just reviewed your content and this is a great improvement over the previous one!
I just have a few things in mind that can be added as changes to make it better:

1. You have an images folder but nowhere in your readme are you mentioning to have a look there or are you adding those images to the readme. 
Kindly do one of the above options.
2. How do you plan on the learners adding their solutions to the questions asked you in the 'Test Yourself' section? (You can either create a `solutions` folder, maybe inside `src` and ask the learners to add their contributions as `.md` files - this is my suggestion, you can think of your own 🛩️ )

Thanks for your generous contribution. It is really important all the sorting algorithms be present in one place for a quick look mainly during interview prep period.

Make these changes and we shall re-review it asap 👍🏽

File: Sorting Algorithms/README.md
Comment (amoghrajesh) [inline Sorting Algorithms/README.md:73]: I think you meant - predecessor* [typo]

**Comment by divya1509** [comment]
I have made the required changes it will be great if you review them.

**Comment by Kevinpaulose05** [comment]
The link for Ur selection sort file in its corresponding activity is wrong

**Review by Kevinpaulose05 — APPROVED** [review]
Great work @divya1509 . Beautiful and immensely technical microbyte. Keep up the work. Another advocate will be reviewing Ur work at the earliest before we merge it.Two things to do **after merging**
1. Open issues for learners so that they can contribute to Ur microbyte
2. Showcase Ur microbyte in slack's microbyte showcase channel using [this](https://github.com/Crio-Bytes/Demo-Repo/blob/master/Maintainer%20Resources/new-microbyte-showcase-template.md)

Happy Contributing!!

**Comment by Kevinpaulose05** [comment]
One last thing from my side. Pls add one article of space time complexity in the references section just so that u don't leave any ambiguity for newbies.

Cheers

**Comment by divya1509** [comment]
@Kevinpaulose05 I have already uploaded an article for space and time complexity and have also fixed the link of selection sort.
Please review that.

**Review by archithdwij — APPROVED** [review]
@divya1509 Very well laid out content. You have created an engaging MicroByte and made sure learners make the most out of it! Cheers!  🎉

**Comment by Kevinpaulose05** [comment]
One last thing regarding opening issues @divya1509 . When u open them try opening two kinds of PRs. One should be about the quiz in the end and they will answer in the answering section. The second issue should be about giving alternatives to ur code(diff lang or enhancement). This code submission they should do by creating a `language` folder inside the `answers` folder. Else it will be a mess later while PR review

**Comment by Kevinpaulose05** [comment]
@all-contributors kindly add @divya1509   for:

- [x] code
- [x] doc
- [x] ideas
- [x] content

**Comment by Ak-Shaw** [comment]
@divya1509 This is some decent content and I know that I'm late to the party. :sweat_smile: 

The others made a nice fuss over your content, and so it made me read through your doc, and I was immersed in it. :100: 
I'm pretty sure that the fuss is well justified. Thank you for your contribution. :heart_decoration: 

Also, kindly showcase your MicroByte in the **#microbyte-showcase** channel over at our slack workspace, using [this](https://github.com/Crio-Bytes/Demo-Repo/blob/master/Maintainer%20Resources/new-microbyte-showcase-template.md) template.

Also, kindly fill [this](https://forms.gle/mE9sB7xR577e2QmG7) form.

**Comment by sudhanshutiwari264** [comment]
@divya1509 I also liked your content and work!
🎉🎉🎉

**Comment by divya1509** [comment]
Thank you all for helping me do this :)

**Pull request merged**
